### PR TITLE
TODO list for tests

### PR DESCRIPTION
**Pull request by v-nikulichev**
As discussed, let's add notes on how to test features we merge into `cpp-worker-gcd` by decomposing `cpp-worker-gcd-mongo`.
I provided an example of a test case `Couple/DC sharing check/Basic setup`. Please have a look at sections _Setup_, _Action_, and _Expected_.

### CODE REVIEW

File: tests/TODO.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+### DESCRIPTION
+
+This file contains a list of test cases and notes on features which should
+be covered with automatic tests.
+
```
Comment (shaitan) [inline tests/TODO.md:5]: This description has to be extended with:
- this file's structure - how it is organized
- schema of test case - HOWTO for adding new test cases and description for understanding existent test cases' structure

File: tests/TODO.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  1. *%Title of test case #1%*. %Brief description of setup, action, and expected
+  result%;
+  2. *%Title of test case #2%*. `...`
+
+### TESTS
+
+#### Couple:
+
+* **DC sharing check**. When option `forbidden_dc_sharing_among_groups` is enabled,
+mastermind checks whether none of pairs of couple's groups has backends in the
```
Comment (shaitan) [inline tests/TODO.md]: What does it mean: "mastermind checks"? Does it use result of this check?

Comment (shaitan) [inline tests/TODO.md]: "whether none of pairs of couple's groups has backends in the same DC" -> "that there is no pair of couple's groups which are located in the same DC."

Comment (v-nikulichev) [inline tests/TODO.md]: Not sure what this comment is about. Grammar correction?
Both sentenses are correct except that 'no pair' is singular, so you should have written "no pair of couple's groups which **is** located in the same DC".

As of none,

> When none is the subject, the verb is either singular or plural depending on what it is referring to.

See http://dictionary.cambridge.org/grammar/british-grammar/no-none-and-none-of

Let's not be grammar geeks anyway :)

File: tests/TODO.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+same DC. The check is made in method `Couple::check_dc_sharing` which is
+invoked by `Couple::update_status`.
+  1. *Basic setup*. There are three groups each of which has one backend. All
```
Comment (shaitan) [inline tests/TODO.md]: What does this test case check, in human words? In facts, it verifies that after some actions some function returns `OK`, but what is the point of this test?

**Comment by shaitan** [comment]
:+1: Looks good, it can be merged. Thank you!

**Pull request merged**